libcontainer/system/proc_test.go

```diff
@@ -52,6 +48,55 @@ func TestParseStat(t *testing.T) {
 	}
 }
 
+func TestParseStatBadInput(t *testing.T) {
+	cases := []struct {
+		desc, input string
+	}{
+		{
+			"no (",
+			"123 ) S 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0",
+		},
+		{
+			"no )",
+			"123 ( S 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0",
+		},
+		{
+			"misplaced ()",
+			"123 )one( S 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0",
+		},
+		{
+			"misplaced empty ()",
+			"123 )( S 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0",
+		},
```

Comment:
I think that cases 1 (aka "( at start") and 3 (aka ") at start") will start failing if we return PID parsing that was removed in the last iteration. So, in fact, this is "invalid PID" or "garbage in field 1" cases, but we don't care about PID.

Another way this can be fixed is looking for ` (` instead of `(`, i.e. add a space, but that will only slow things down.

Case 2 (aka "( at end") seems legit from the current parser's POV as it only cares about the field within `()` (aka Name), the field right after (aka State), and the Start Time (22nd field).

Case 4 is seen as error as there are not enough fields after `)`.

Now, there are more situations in which the current parser won't detect an error. For example, if any field from 4 to 52 is not a number, or empty, or (for some of them) negative, etc. -- we don't detect any of that.

We do not want to validate the input here, as we trust the kernel. We only want to
 - extract the fields we're interested in;
 - be sure to not panic in any circumstances.
 
 If you can think of any cases where the current code would panic, that'd be great.

Reply:
aaaand I just did:

```
                {
                        "a tad short",
                        "1234 (cmd) ",
                },
```

Reply:
Improved the parser to bail early on an incorrect input, and documented it a bit more.

Reply:
PTAL @thaJeztah